perf/perf_fuzzer.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM
+# Author:Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
+#
+
+import platform
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import archive, build, process, distro, genio
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Perffuzzer(Test):
+
+    """
+    This test run the perf fuzzer, which fuzz's the Linux perf_event
+    system call interface.
+    Source: https://github.com/deater/perf_event_tests/fuzzer
+    (http://web.eece.maine.edu/~vweaver/projects/perf_events/fuzzer/)
+    This test might crash the kernel
+    :avocado: tags=privileged
+    """
+
+    @staticmethod
+    def run_cmd_out(cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True, sudo=True)
+
+    def setUp(self):
+        '''
+        Install the packages
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in detected_distro.name:
+            kernel_ver = platform.uname()[2]
+            deps.extend(['linux-tools-common', 'linux-tools-%s'
+                         % kernel_ver])
+        # FIXME: "redhat" as the distro name for RHEL is deprecated
+        # on Avocado versions >= 50.0.  This is a temporary compatibility
+        # enabler for older runners, but should be removed soon
+        elif detected_distro.name in ['rhel', 'SuSE', 'fedora', 'redhat']:
+            deps.extend(['perf'])
+        else:
+            self.cancel("Install the package for perf supported by %s"
```

Comment:
can this message  will be having more sense 
            `self.cancel("Test not supported on  %s"% detected_distro.name)`

Reply:
Here is the `perf` package is missing, the right error is to complain about the missing package instead of failing with test not supported.

Reply:
sure make sense so i think message can be rephrase as 
```Perf package installation not supported by %s or can be issue with software repository please check  "% detected_distro.name ```